lib/src/presentation/bloc/cards_bloc/cards_bloc_impl.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import 'package:flutter/cupertino.dart';
+
+import '../../../core/parameter/card_request_params.dart';
+import '../../../core/util/constants.dart';
+import '../../../domain/entity/card_event.dart';
+import '../../../core/usecases/usecase.dart';
+import 'cards_bloc.dart';
+
+class CardsBlocImpl extends CardsBloc {
+  final Usecase _getCardsUsecase;
+  CardsRequestParams _cardsRequestParams;
+  CardEvent _cardEvent;
+
+  CardsBlocImpl(this._getCardsUsecase, this._cardsRequestParams, this._cardEvent);
+
+  final StreamController<CardEvent> _cardsStreamController =
+      StreamController();
+
+  @override
+  Stream<CardEvent> getStream() => _cardsStreamController.stream;
+
+  @override
+  void dispose() {
+    _cardsStreamController.close();
+  }
+
+  @override
+  Future<void> getFilteredCards(endpoint) async {
+    _cardEvent.status = Status.loading;
+    _cardsStreamController.sink.add(_cardEvent);
+    _cardEvent.cards.clear();
+    if (endpoint != null) {
```

Comment:
What it the scenario that allows an endpoint be null?

Reply:
You are right, there is no scenario